adaptive_breakpoints/example/lib/main.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:adaptive_breakpoints/adaptive_breakpoints.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        body: Column(
+          children: [
+            AdaptiveContainer(
+              windowLimit: AdaptiveWindowType.xs,
+              child: Text('This is a extra small window'),
+            ),
+            AdaptiveContainer(
+              windowLimit: AdaptiveWindowType.s,
+              child: Text('This is a small screen'),
+            ),
+            AdaptiveContainer(
+              windowLimit: AdaptiveWindowType.m,
+              child: Text('This is a medium window'),
+            ),
+            AdaptiveContainer(
+              windowLimit: AdaptiveWindowType.l,
+              child: Text('This is a large window'),
+            ),
+            AdaptiveContainer(
+              windowLimit: AdaptiveWindowType.xl,
+              child: Text('This is a extra large window'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// This example application demonstrates how to use
+/// [`adaptive_breakpoints`](https://pub.dev/packages/adaptive_breakpoints)
+/// within a simple Flutter app.
+class AdaptiveContainer extends StatelessWidget {
+  final AdaptiveWindowType windowLimit;
+  final Widget child;
+
+  const AdaptiveContainer({
+    @required this.windowLimit,
+    this.child,
+  })  : assert(windowLimit != null);
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        BreakpointSystemEntry entry = getBreakpointEntry(context);
+        if (entry.window == windowLimit) {
+          return Container(
+            constraints: BoxConstraints(
+              minWidth: entry.window.widthRangeValues.start,
+              maxWidth: entry.window.widthRangeValues.end,
+            ),
+            width: MediaQuery.of(context).size.width - (entry.margins * 2),
+            margin: EdgeInsets.symmetric(horizontal: entry.margins),
+            child: child,
+          );
+        } else {
+          return SizedBox();
+        }
+      },
+    );
+  }
+}
```

Comment:
This code is repeated in the test. It's probably pretty useful in the context of this package. Consider including it in the package API, with documentation for windowLimit

Reply:
I have thought of that too.

I think it would be useful to add the container type here. But it might be in a separate package that handles adaptive experiences in Flutter.

Reply:
I think it'd be best suited in this package since it relies on AdaptiveWindowType. Then users can avoid the width, constraints & margin boilerplate

Reply:
I ended up adding the variable and added it to the package. Since this is pre 1.0.0 we can remove it in the future and add it to a separate package if required.